src/utils/SpokeUtils.ts

```diff
@@ -240,6 +240,24 @@ function getV3RelayHash(relayData: V3RelayData, destinationChainId: number): str
   );
 }
 
+export function getV3RelayHashFromEvent(e: V3Deposit | V3Fill | SlowFillRequest): string {
+  const relayData: V3RelayData = {
+    depositor: e.depositor,
+    recipient: e.recipient,
+    exclusiveRelayer: e.exclusiveRelayer,
+    inputToken: e.inputToken,
+    outputToken: e.outputToken,
+    inputAmount: e.inputAmount,
+    outputAmount: e.outputAmount,
+    originChainId: e.originChainId,
+    depositId: e.depositId,
+    fillDeadline: e.fillDeadline,
+    exclusivityDeadline: e.exclusivityDeadline,
+    message: e.message,
+  };
+  return getV3RelayHash(relayData, e.destinationChainId);
```

Comment:
`SlowFillRequest` is currently just a 1:1 match for `V3RelayData`, so it lacks `destinationChainId. I'm not sure why tsc isn't complaining about it. I've proposed to add `destinationChainId` here: https://github.com/across-protocol/sdk-v2/pull/536

![common](https://github.com/across-protocol/sdk-v2/assets/108695806/e8e3dcde-17f0-4685-976e-1863bf5052e2)

![relayData](https://github.com/across-protocol/sdk-v2/assets/108695806/6c1c047e-6db2-48f9-b451-881cbdd06194)

![slowFillrequest](https://github.com/across-protocol/sdk-v2/assets/108695806/70306a83-a812-4900-a352-892a25e32b35)

Reply:
Have merged in the above commit, so we should be OK now.

Reply:
Should we instead just pass in `destinationChainId` into this function?

Reply:
https://github.com/across-protocol/sdk-v2/pull/535/commits/78a64ab0670428446d0506933a8346fa6179c4b9

Reply:
I realized that even though `V3FilledRelay` has a `destinationChainId`, it doesn't get populated by the `SpokePoolClient` because this param doesn't exist in the onchain event. I think the issue is the type casting we do here overrides the typeguard 
![Screenshot 2024-02-06 at 13 43 57](https://github.com/across-protocol/sdk-v2/assets/9457025/0d2fb2f6-2d96-46f3-8c34-c7b7cd0d5aa3)

Reply:
I think you need this fix https://github.com/across-protocol/sdk-v2/commit/d429f1f93f5bbfd79416cf9c644c82c726d6390d

Reply:
Good catch - this is of course one of the pitfalls of the prevailing method of unpacking events in the SpokePoolClient. I've applied the corresponding fix to v3types and added tests to verify it.

https://github.com/across-protocol/sdk-v2/pull/501/commits/960e6f01fc3aaaad6b8b9ff8aa76873845b24b14